src/main/java/appeng/client/render/cablebus/FacadeBuilder.java

```diff
@@ -19,370 +19,364 @@
-					tintColor |= color & 0xFF000000;
-					builder.setColor( tintColor );
-				}
-				else
+			List<BakedQuad> modelQuads = new ArrayList<>();
+			//If we are forcing transparent facades, fake the render layer, and grab all quads.
+			if( transparent )
+			{
+				for( BlockRenderLayer forcedLayer : BlockRenderLayer.values() )
 				{
-					builder.setColor( color );
+					//Check if the block renders on the layer we want to force.
+					if( blockState.getBlock().canRenderInLayer( blockState, forcedLayer ) )
+					{
+						//Force the layer and gather quads.
+						ForgeHooksClient.setRenderLayer( forcedLayer );
+						modelQuads.addAll( gatherQuads( model, blockState, rand ) );
+					}
 				}
-				builder.setTexture( facing, spriteAndTint.sprite );
+
+				//Reset.
+				ForgeHooksClient.setRenderLayer( layer );
 			}
 			else
 			{
-				builder.setColor( color );
-				builder.setTexture( facing, this.facadeTexture );
+				modelQuads.addAll( gatherQuads( model, blockState, rand ) );
 			}
-		}
-
-		builder.setDrawFaces( facadeState.getOpenFaces() );
-
-		AxisAlignedBB primaryBox = getFacadeBox( side, thinFacades );
 
-		Vector3f min = new Vector3f( (float) primaryBox.minX * 16, (float) primaryBox.minY * 16, (float) primaryBox.minZ * 16 );
-		Vector3f max = new Vector3f( (float) primaryBox.maxX * 16, (float) primaryBox.maxY * 16, (float) primaryBox.maxZ * 16 );
-
-		if( busBounds == null )
-		{
-			// Adjust the facade for neighboring facades so that facade cubes dont overlap with each other
-			if( side == EnumFacing.NORTH || side == EnumFacing.SOUTH )
+			//No quads.. Cool, next!
+			if( modelQuads.isEmpty() )
 			{
-				if( facades.containsKey( EnumFacing.UP ) )
-				{
-					max.y -= thickness;
-				}
-
-				if( facades.containsKey( EnumFacing.DOWN ) )
-				{
-					min.y += thickness;
-				}
+				continue;
 			}
-			else if( side == EnumFacing.EAST || side == EnumFacing.WEST )
-			{
-				if( facades.containsKey( EnumFacing.UP ) )
-				{
-					max.y -= thickness;
-				}
 
-				if( facades.containsKey( EnumFacing.DOWN ) )
-				{
-					min.y += thickness;
-				}
+			//Grab out pipeline elements.
+			QuadClamper clamper = pipeline.getElement( "clamper", QuadClamper.class );
+			QuadFaceStripper edgeStripper = pipeline.getElement( "face_stripper", QuadFaceStripper.class );
+			QuadTinter tinter = pipeline.getElement( "tinter", QuadTinter.class );
+			QuadCornerKicker kicker = pipeline.getElement( "corner_kicker", QuadCornerKicker.class );
 
-				if( facades.containsKey( EnumFacing.SOUTH ) )
-				{
-					max.z -= thickness;
-				}
+			//Set global element states.
 
-				if( facades.containsKey( EnumFacing.NORTH ) )
+			//calculate the side mask.
+			int facadeMask = 0;
+			for( Entry<EnumFacing, FacadeRenderState> ent : facadeStates.entrySet() )
+			{
+				EnumFacing s = ent.getKey();
+				if( s.getAxis() != side.getAxis() )
 				{
-					min.z += thickness;
+					FacadeRenderState otherState = ent.getValue();
+					if( !otherState.isTransparent() )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)